src/Particle/WalkerConfigurations.h

```diff
@@ -0,0 +1,201 @@
+ * - LivingFraction fraction of walkers alive each step.
+ */
+template<typename T>
+struct MCDataType
+{
+  T NumSamples;
+  T RNSamples;
+  T Weight;
+  T Energy;
+  T AlternateEnergy;
+  T Variance;
+  T R2Accepted;
+  T R2Proposed;
+  T LivingFraction;
+};
+
+/** A set of light weight walkers that are carried between driver sections and restart
+ */
+class WalkerConfigurations
+{
+public:
+  /// walker type
+  using Walker_t         = Walker<QMCTraits, PtclOnLatticeTraits>;
+  using FullPrecRealType = QMCTraits::FullPrecRealType;
+  ///container type of Walkers
+  using WalkerList_t = std::vector<Walker_t*>;
+  /// FIX: a type alias of iterator for an object should not be for just one of many objects it holds.
+  using iterator = WalkerList_t::iterator;
+  ///const_iterator of Walker container
+  using const_iterator = WalkerList_t::const_iterator;
+
+  /** starting index of the walkers in a processor group
+   *
+   * WalkerOffsets[0]=0 and WalkerOffsets[WalkerOffsets.size()-1]=total number of walkers in a group
+   * WalkerOffsets[processorid+1]-WalkerOffsets[processorid] is equal to the number of walkers on a processor,
+   * i.e., W.getActiveWalkers().
+   * WalkerOffsets is added to handle parallel I/O with hdf5
+   */
+  std::vector<int> WalkerOffsets;
+
+  MCDataType<FullPrecRealType> EnsembleProperty;
+
+  WalkerConfigurations();
+  ~WalkerConfigurations();
+  WalkerConfigurations(const WalkerConfigurations&) = delete;
+  WalkerConfigurations& operator=(const WalkerConfigurations&) = delete;
+  WalkerConfigurations(WalkerConfigurations&&)                 = default;
+  WalkerConfigurations& operator=(WalkerConfigurations&&) = default;
+
+  /** create numWalkers Walkers
+   *
+   * Append Walkers to WalkerList.
+   */
+  void createWalkers(int numWalkers, size_t numPtcls);
+  /** create walkers
+   * @param first walker iterator
+   * @param last walker iterator
+   */
+  void createWalkers(iterator first, iterator last);
+  /** copy walkers
+   * @param first input walker iterator
+   * @param last input walker iterator
+   * @param start first target iterator
+   *
+   * No memory allocation is allowed.
+   */
+  void copyWalkers(iterator first, iterator last, iterator start);
+
+  /** destroy Walkers from itstart to itend
+   *@param first starting iterator of the walkers
+   *@param last ending iterator of the walkers
+   */
+  iterator destroyWalkers(iterator first, iterator last);
+
+  /** destroy Walkers
+   *@param nw number of walkers to be destroyed
+   */
+  void destroyWalkers(int nw);
+
+  ///clean up the walker list and make a new list
+  void resize(int numWalkers, size_t numPtcls);
+
+  ///return the number of active walkers
+  inline size_t getActiveWalkers() const { return WalkerList.size(); }
+  ///return the total number of active walkers among a MPI group
+  inline size_t getGlobalNumWalkers() const { return GlobalNumWalkers; }
+  ///return the total number of active walkers among a MPI group
```

Comment:
This collides with some of MCPopulations responsibilities and I think that the GlobalNumWalkers state is out of scope for this class which I would like to see as a clean data object of rank scope.  We shouldn't be shy about cleaning this up in the future.